### PR TITLE
example/with-typescript-plugin

### PR DESCRIPTION
**Pull request by deptno**
#3694

### CODE REVIEW

File: examples/with-typescript2/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.js
```
Comment (sergiodxa) [inline examples/with-typescript2/.gitignore]: Not an expert of TS but do we still need to ignore .js files if we are using the webpack loader?

File: examples/with-typescript2/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+Install it and run:
+ 
+```
+npm install
+npm run dev
+```  
+
+Output JS files are aside the related TypeScript ones.
```
Comment (sergiodxa) [inline examples/with-typescript2/README.md]: Is this still true?

File: examples/with-typescript2/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "with-typescript2",
```
Comment (sergiodxa) [inline examples/with-typescript2/package.json]: I think we can either remove the old example and use this or name it `with-typescript-plugin`, what do you think @timneutkens?

Comment (sergiodxa) [reply examples/with-typescript2/package.json]: Maybe maintain the old one for compatibility with old Next.js versions?

File: examples/with-typescript2/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "compileOnSave": false,
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
+    "jsx": "preserve",
```
Comment (sergiodxa) [inline examples/with-typescript2/tsconfig.json]: Do we still need to preserve JSX?

**Review by sergiodxa — CHANGES_REQUESTED** [review]

**Comment by sergiodxa** [comment]
@deptno awesome (and super fast) work!

File: examples/with-typescript2/README.md (hunk 2 above)
Comment (deptno) [reply examples/with-typescript2/README.md]: I removed that line

File: examples/with-typescript2/tsconfig.json (hunk 4 above)
Comment (deptno) [reply examples/with-typescript2/tsconfig.json]: this option from <https://github.com/zeit/next-plugins/tree/master/packages/next-typescript>

I’ve tested styled-jsx with jsx option removal,
It works well, I removed from code

File: examples/with-typescript2/package.json (hunk 3 above)
Comment (deptno) [reply examples/with-typescript2/package.json]: replace with-typescript2 -> with-typescript-plugin

**Comment by timneutkens** [comment]
yeah please make this `with-typescript` the old example was a hack.

**Review by deptno — COMMENTED** [review]
done

**Review by sergiodxa — APPROVED** [review]

**Comment by kachkaev** [comment]
Hi guys,

I just used your example to start a new project and noticed this error message from in VSCode:

![[ts] Cannot use JSX unless the '--jsx' flag is provided.](https://user-images.githubusercontent.com/608862/36341639-3f7de102-13e9-11e8-925d-45a6bc50f448.png)

After a bit of googling, I could solve this by adding `"jsx": "preserve"` to `tsconfig.json` and restarting the VSCode's TS server.

What made you remove this key from the config? Should we bring it back?

**Comment by timneutkens** [comment]
@kachkaev the example should reflect: https://github.com/zeit/next-plugins/tree/master/packages/next-typescript#usage, feel free to create a pull request 🙏

**Pull request merged**